pkg/roachprod/roachprodutil/identity.go

```diff
@@ -7,129 +7,129 @@ package roachprodutil
 
 import (
 	"context"
-	"io"
+	"net/http"
 	"os"
-	"path/filepath"
-	"strings"
+	"time"
 
-	"cloud.google.com/go/storage"
+	"github.com/binxio/gcloudconfig"
+	"github.com/cockroachdb/cockroach/pkg/util/httputil"
 	"github.com/cockroachdb/errors"
 	"golang.org/x/oauth2"
-	"google.golang.org/api/idtoken"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/impersonate"
 	"google.golang.org/api/option"
 )
 
-var (
-	userHome, _ = os.UserHomeDir()
-	// serviceAccountCredFile is where the service account credentials are
-	// stored that enable Google IDP auth.
-	serviceAccountCredFile = filepath.Join(userHome, ".roachprod", "service-account-secrets")
+const (
+	// CredentialsEnvironmentVariable is the environment variable that takes
+	// precedence over the GCP default credentials mechanism.
+	CredentialsEnvironmentVariable = "GOOGLE_EPHEMERAL_CREDENTIALS"
+	cloudPlatformScope             = "https://www.googleapis.com/auth/cloud-platform"
 )
 
-// FetchedFrom indicates where the credentials have been fetched from.
-// this helps in debugging and testing.
-type FetchedFrom string
+// IAPTokenSourceIface is an interface that defines the methods
+// for the IAPTokenSource.
+type IAPTokenSourceIface interface {
```

Comment:
nit: this being `IAPTokenSource` and the implementation below being `IAPTokenSourceImpl` is more idiomatic with the rest of the roachtest/roachprod code